src/deepsparse/pipeline_logger.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+__all__ = ["PipelineLogger"]
+
+
+class PipelineLogger(ABC):
+    """
+    Generic PipelineLogger abstract class meant to define interfaces
+    for the loggers that support various monitoring services APIs.
+
+    :param identifier: The name of the monitoring service that the
+        PipelineLogger uses to log the inference data
+    """
+
+    def __init__(self, identifier: str):
+        self._identifier = identifier
+
+    @property
```

Comment:
Related to my other comment.
Let's make this an abstract property for subclasses to implement, then it doesn't have to given to this class' constructor. And then the prometheus one can just define it as a property instead of having a global variable for it.